src/components/PostDetails.tsx

```diff
@@ -1,118 +1,125 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment, CommentData } from '../types/Comment';
+
+type Props = {
+  post: Post;
+  comments: Comment[] | null;
+  hasCommentsError: boolean;
+  removeComment: (commentId: number) => void;
+  addComment: (comment: CommentData) => void;
+  isCommentAdditionError: boolean;
+  isCommentAdditionLoading: boolean;
+};
+
+export const PostDetails: React.FC<Props> = React.memo(({
+  post,
+  comments,
+  hasCommentsError,
+  removeComment,
+  addComment,
+  isCommentAdditionError,
+  isCommentAdditionLoading,
+}) => {
+  const [isFormActive, setIsFormActive] = useState(false);
+
+  useEffect(() => {
+    return () => {
```

Comment:
Remove the useEffect block on lines 29-33, as it is not necessary to reset the state when the post changes.

Reply:
I don't think so. It is necessary to reset the state to show the new post details form, and this is a possible implementation.